docs/java-api/client.asciidoc

```diff
@@ -23,9 +23,9 @@ that connects to a cluster.
 *Important:*
 
 ______________________________________________________________________________________________________________________________________________________________
-Please note that you are encouraged to use the same version on client
-and cluster sides. You may hit some incompatibility issues when mixing
-major versions.
+You may use the Java client as long as the major version matches. You should upgrade to use a matching version at your earliest convenience. You should upgrade the client before upgrading the cluster. You may not use a client that differs across major versions.
```

Comment:
I know I typed this myself but I might reverse the order of the two should sentences - I think it'd read more like a recommendation.